spec/requests/topics_controller_spec.rb

```diff
@@ -2380,6 +2380,31 @@ def fabricate_topic(user, category = nil)
       expect(response.status).to eq(404)
     end
 
+    it "handles pagination correctly with deleted posts" do
+      topic_with_posts = Fabricate(:topic)
+      posts_to_delete = []
+
+      24.times do |i|
+        post = Fabricate(:post, topic: topic_with_posts)
+        # mark every other post for deletion
+        posts_to_delete << post if i.even?
+      end
+
+      posts_to_delete.each(&:trash!)
```

Comment:
I wonder, can we speed up this test by fabricating with these properties  for even posts: 

deleted_at: ..., deleted_by_id: ...